### PR TITLE
Fix ECS Fargate processes not being associated with containers

### PR DESCRIPTION
**Pull request by sblumenthal**
### What does this PR do?

Fixes a bug where sometimes the process data generated for an ECS Fargate container is missing the container tags

### Motivation

https://datadoghq.atlassian.net/browse/CONS-6242

### Additional Notes

It appears that the problem was due to regex parsing, while debugging I found instances where the fargate container ID was {32}-{9}, but we were explicitly only looking at {32}-{10}. I cannot find any information on the definition of the fargate container ID format, so while this addresses the situation we came across in the support ticket, I do not know if there are other cases we have not run into yet.

### Describe how to test/QA your changes

I was able to reliably replicate this issue using the python container using the following task container definition:
```
        {
            "name": "api",
            "image": "public.ecr.aws/docker/library/python",
            "essential": false,
            "command": [
                "python",
                "-m",
                "http.server",
                "8080"
            ]
        }
```
Add this container definition as another container [here](https://github.com/DataDog/container-agent-envs-doc/blob/main/create_test_env/2_3_agent_sidecar_dummy_app_management/ecs-fargate/tasks/redis.json#L85), after the redis container definition, and deploy to an ECS fargate cluster

Verify that the container shows up on the live containers page, and that the process data shows up attached to this container, like so:
image

Also, to make sure the original bug that we ran into was being hit, log into the AWS console and navigate to the tasks page for you cluster, find the task that contains the python container, and validate that the container runtime ID matches the format that caused the issue in the first place:
image

The above container reliably triggered the issue, across multiple clusters. However, if it does not result in a container ID that is one character short, then it is worth noting that the QA for this has already been done in a cluster where we were able to reproduce the issue, and has been validated

### CODE REVIEW

**Review by AliDatadog — APPROVED** [review]

File: pkg/util/cgroups/pid_mapper_test.go [hunk 1]
```diff
@@ -216,6 +228,11 @@ func TestIdentiferFromCgroupReferences(t *testing.T) {
 			fileContent: ecsFargateCgroup,
 			expectedID:  "8474ac4cec7a4f488834b00591271ec3-3054012820",
 		},
+		{
+			name:        "ecs fargate shorter",
+			fileContent: ecsFargateCgroup,
```
Comment (AliDatadog) [inline pkg/util/cgroups/pid_mapper_test.go:233]:
```suggestion
			fileContent: ecsFargateCgroupShort,
```

Comment (sblumenthal) [reply pkg/util/cgroups/pid_mapper_test.go]: Yup, just noticed failing jobs, my bad

**Review by robertjli — APPROVED** [review]

**Review by rtrieu — APPROVED** [review]
Looks good!

**Pull request merged**